demisto_sdk/commands/common/logger.py

```diff
@@ -1,100 +1,173 @@
-import itertools
-import logging
-import logging.config
-import os.path
+import os
 import platform
-import re
 import sys
-from logging.handlers import RotatingFileHandler
 from pathlib import Path
-from typing import Any, Dict, List, Optional, Union
+from typing import Iterable, Optional, Union
+
+import loguru  # noqa: TID251 # This is the only place where we allow it
 
-# NOTE: Do not add internal imports here, as it may cause circular imports.
 from demisto_sdk.commands.common.constants import (
-    DEMISTO_SDK_LOG_FILE_COUNT,
     DEMISTO_SDK_LOG_FILE_PATH,
     DEMISTO_SDK_LOG_FILE_SIZE,
     DEMISTO_SDK_LOG_NO_COLORS,
     DEMISTO_SDK_LOG_NOTIFY_PATH,
+    DEMISTO_SDK_LOGGING_SET,
     LOG_FILE_NAME,
     LOGS_DIR,
-    STRING_TO_BOOL_MAP,
+)
+from demisto_sdk.commands.common.string_to_bool import (
+    string_to_bool,  # See the comment in string_to_bool's implementation
 )
 
-logger: logging.Logger = logging.getLogger("demisto-sdk")
+FILE_FORMAT = (
+    "{time:YYYY-MM-DD HH:mm:ss.SSS} | {level} | {message} @ {file}:{function}:{line}"
+)
+CONSOLE_FORMAT = "{message}"
 
+DEFAULT_FILE_THRESHOLD = "DEBUG"
+DEFAULT_CONSOLE_THRESHOLD = "INFO"
 
-def environment_variable_to_bool(
-    variable_name: str, default_value: bool = False
-) -> bool:
-    """
-    Check if the environment variable is set and is a valid boolean value.
-    If it is not set or is not a valid boolean value, return the default value.
+DEFAULT_FILE_SIZE = 1 * (1024**2)  # 1 MB
+DEFAULT_FILE_COUNT = 10
 
-    Args:
-        variable_name (str): The name of the environment variable.
-        default_value (bool): A default value to return if the environment variable is not set or is invalid.
+global logger
+logger = loguru.logger  # all SDK modules should import from this file, not from loguru
+logger.disable(None)  # enabled at setup_logging()
 
-    Returns:
-        bool: The environment variable value if it is set and is a valid boolean value, otherwise the default value.
-    """
-    env_var = os.getenv(variable_name)
 
-    if not env_var:
-        return default_value
+def _setup_neo4j_logger():
+    import logging  # noqa: TID251 # special case, to control the neo4j logging
+
+    neo4j_log = logging.getLogger("neo4j")
+    neo4j_log.setLevel(logging.CRITICAL)
+
+
+def calculate_log_size() -> int:
+    if env_var := os.getenv(DEMISTO_SDK_LOG_FILE_SIZE):
+        try:
+            return int(env_var)
+
+        except (TypeError, ValueError):
+            logger.warning(
+                f"non-integer log-size value ({env_var}). Defaulting to {DEFAULT_FILE_SIZE}B."
+            )
+    return DEFAULT_FILE_SIZE
+
+
+def calculate_rentation() -> int:
+    if env_var := os.getenv("DEMISTO_SDK_LOG_FILE_COUNT"):
+        try:
+            return int(env_var)
+        except (TypeError, ValueError):
+            logger.warning(
```

Comment:
not sure it should be a warning, isn't it the common case that this var will not be in environment?

Reply:
In line 58, we check for its existence (`if env_var := os.getenv("DEMISTO_SDK_LOG_FILE_COUNT"):`)
If it's falsey, we go to line 65, and return the default.